version: '2.4'
services:
  pg:
    image: postgres:9.6
    restart: always

    environment:
      POSTGRES_PASSWORD: postgres
      POSTGRES_USER: postgres
      POSTGRES_DB: face_index
    healthcheck:
        test: ["CMD-SHELL", "pg_isready -U postgres"]
        interval: 2s
        timeout: 2s
        retries: 10
    ports:
      - "5432:5432"
    command:
        "postgres -c max_connections=128 -c shared_buffers=4GB -c effective_cache_size=4GB -c work_mem=256MB"

  face_index:
    mem_limit: 16G
    build:
      context: .
      dockerfile: docker/Dockerfile
    restart: always
    tty: true
    ports:
      - "4445:4445"
    environment:
      PYTHONUNBUFFERED: 1
      FACE_INDEX_PORT: 4445
      DB_URL: postgresql://postgres:postgres@pg:5432/face_index
